dashboard/modules/job/job_description.py

```diff
@@ -0,0 +1,58 @@
+import attr
+from attr.validators import instance_of, in_, deep_mapping
+from ray.core.generated import common_pb2
+
+
+@attr.s(repr=False, slots=True, hash=True)
+class _AnyValidator(object):
+    def __call__(self, inst, name, value):
+        pass
+
+    def __repr__(self):
+        return "<any validator for any type>"
+
+
+def any_():
+    return _AnyValidator()
+
+
+@attr.s(kw_only=True, slots=True)
+class RuntimeEnv:
+    # The url to download the job package archive. The archive format is
+    # one of “zip”, “tar”, “gztar”, “bztar”, or “xztar”. Please refer to
+    # https://docs.python.org/3/library/shutil.html#shutil.unpack_archive
+    workingDir = attr.ib(type=str, validator=instance_of(str))
+
+
+@attr.s(kw_only=True, slots=True)
```

Comment:
Btw, it would be great if we could use `Pydantic` instead of `attr` for type validation (that's the modern standard for python web apps and we use it in other places as well). Not a blocker for this PR.

Reply:
Snake-case will break google JSON style. If that's OK, I am glad to change the style to snake-case.

`attr` is used by the aiohttp, so I choose it for type validation. There are many similar libraries:
- `attrs` (3.5k stars) used by aiohttp
- `traitlets` (421 stars) used by jupyter notebook.
- `pydantic` (6.3k stars) used by FastAPI and some subprojects of jupyter notebook.

The `pydantic` supports JSON better, but `attrs` also meet our needs.

Reply:
> I think it's better to break google style in order to make the runtime_env API pythonic. I'd also prefer to use pydantic here just because we use it widely in other parts of the code (like Ray Serve) and internally at anyscale so it'll be easier to maintain

OK. I will use `pydantic` instead of `attrs`, and make the JSON keys snake-case.